### PR TITLE
Allow to upload, dump, validate only jobs which are assigned to the user

### PR DESCRIPTION
**Pull request by alexanderGerbik**
Extra feature for https://p.daedalean.ai/T6020

When multiple annotators are working on the same task, it's confusing when dump and validate processes all jobs (even those which are not assigned to the current user) and it's not safe to upload to the whole task as it might overwrite other annotators' data.

### CODE REVIEW

File: cvat-core/src/server-proxy.js [hunk 1]
```diff
@@ -249,12 +249,19 @@
                 }
             }
 
-            async function validateTask(id) {
+            async function validateTask(id, jobs) {
                 const { backendAPI } = config;
 
+                let query = "";
+                if (jobs !== null) {
```
Comment (gyakushev) [inline cvat-core/src/server-proxy.js]: better use `if(jobs)` here: `undefined` is also not equal to `null`

File: cvat-core/src/server-proxy.js [hunk 2]
```diff
@@ -249,12 +249,19 @@
...
+                    const queryParams = new URLSearchParams();
+                    queryParams.append("jobs", jobs);
```
Comment (gyakushev) [inline cvat-core/src/server-proxy.js:258]: nit: it is useful to keep links human-readable
you can use `jobs.join()` here

File: cvat-core/src/server-proxy.js [hunk 3]
```diff
@@ -546,17 +553,22 @@
...
                 const { backendAPI } = config;
 
                 let annotationData = new FormData();
                 annotationData.append('annotation_file', file);
 
                 return new Promise((resolve, reject) => {
                     async function request() {
+                        const query = new URLSearchParams();
+                        query.append("format", format);
+                        if (jobs != null && session !== "job") {
```
Comment (gyakushev) [inline cvat-core/src/server-proxy.js]: ditto

File: cvat-ui/src/components/task-page/task-page.tsx [hunk 4]
```diff
@@ -71,14 +83,23 @@ class TaskPageComponent extends React.PureComponent<Props> {
             );
         }
 
+        const mineJobs = onlyMine ? task.instance.jobs.filter(j => (j.assignee || {}).id === me.id).map(j => j.id) : null;
```
Comment (gyakushev) [inline cvat-ui/src/components/task-page/task-page.tsx]: split this statement into multiple lines

File: cvat-ui/src/components/task-page/top-bar.tsx [hunk 5]
```diff
@@ -19,22 +21,33 @@ import { MenuIcon } from 'icons';
...
     const { id } = taskInstance;
 
     return (
-        <Row className='cvat-task-top-bar' type='flex' justify='space-between' align='middle'>
+        <Row className='cvat-task-top-bar' type='flex' justify='end' align='middle'>
             <Col>
                 <Text className='cvat-title'>{`Task details #${id}`}</Text>
             </Col>
             <Col>
+                <Tooltip title='Show, dump, upload, validate only mine jobs'>
```
Comment (gyakushev) [inline cvat-ui/src/components/task-page/top-bar.tsx]: `only my jobs`

File: cvat-ui/src/styles.scss [hunk 6]
```diff
@@ -18,6 +18,10 @@ hr {
     margin: 10% 25%;
 }
 
+.cvat-text-color {
```
Comment (gyakushev) [inline cvat-ui/src/styles.scss:21]: the class name sounds too generic: does it really apply to all text in CVAT?
is there a specific meaning for this class?

**Review by gyakushev — APPROVED** [review]
Very good. No blockers.

File: cvat-ui/src/styles.scss (hunk 6 above)
Comment (alexanderGerbik) [reply cvat-ui/src/styles.scss:21]: I've removed that class in my previous PR - https://github.com/daedaleanai/cvat/pull/11/files?#diff-3f857da70921ad42793f9f8e4eddf5e9L21. I've reverted some minor changes and have to add the class back. I want to leave it as it was before to not to diverge a lot from the upstream.

File: cvat-core/src/server-proxy.js (hunk 2 above)
Comment (alexanderGerbik) [reply cvat-core/src/server-proxy.js:258]: Do you mean not using `URLSearchParams` at all and formatting `jobs` right into url? That way:
``` `${backendAPI}/tasks/${id}/validate?jobs=${jobs.join(',')}` ```

Comment (gyakushev) [reply cvat-core/src/server-proxy.js:258]: You can still use `URLSearchParams` if you wish. But the param value can be `jobs.join()` to achieve the same effect.

**Pull request merged**